test/es6/forloops-per-iteration-bindings.js

```diff
@@ -241,12 +241,12 @@ var tests = [
     {
         name: "for loops allow const loop variables but cannot assign to them anywhere including the increment expression", // so they're kinda useless
         body: function () {
-            assert.throws(function () { eval("for (const x = 0; x++ < 3; ) { }"); }, SyntaxError, "assignment to const known at parse time in the test expression", "Assignment to const");
-            assert.throws(function () { for (const x = 0; eval("x++") < 3; ) { } }, ReferenceError, "assignment to const known at run time in the test expression", "Assignment to const");
-            assert.throws(function () { eval("for (const x = 0; x < 3; x += 1) { }"); }, SyntaxError, "assignment to const known at parse time in the increment expression", "Assignment to const");
-            assert.throws(function () { for (const x = 0; x < 3; eval("x += 1")) { } }, ReferenceError, "assignment to const known at run time in the increment expression", "Assignment to const");
-            assert.throws(function () { eval("for (const x = 0; x < 3; ) { x += 1; }"); }, SyntaxError, "assignment to const known at parse time in the body", "Assignment to const");
-            assert.throws(function () { for (const x = 0; x < 3; ) { eval("x += 1"); } }, ReferenceError, "assignment to const known at run time in the body", "Assignment to const");
+            assert.throws(function () { eval("for (const x = 0; x++ < 3; ) { }"); }, TypeError, "assignment to const known at parse time in the test expression", "Assignment to const");
```

Comment:
These wordings seem okay in describing the differences between the lines ("known" != "thrown"), but we can certainly change them if there's a better way.